File: Editor/src/main/java/org/turbomedia/turboedit/editor/renderer/RenderServerConnection.java
package org.turbomedia.turboedit.editor.renderer;

import org.turbomedia.turboedit.editor.events.EventSystem;
import org.turbomedia.turboedit.editor.events.EventType;
import org.turbomedia.turboedit.editor.events.RendererServerStatusEventData;

public final class RenderServerConnection {
    private final RenderServerEntry entry;
    private RendererClient client;
    private ConnectionStatus status = ConnectionStatus.CONNECTING;

    public RenderServerConnection(RenderServerEntry entry, RendererClient client) {
        this.entry = entry;
        this.client = client;
    }

    public RenderServerEntry entry() {
        return entry;
    }

    public RendererClient client() {
        return client;
    }

    public void client(RendererClient client) {
        this.client = client;
    }

    public ConnectionStatus status() {
        return status;
    }

    public void status(ConnectionStatus status) {
        status(status, true);
    }

    public void status(ConnectionStatus status, boolean callEvent) {
        if (callEvent) {
            var eventData = new RendererServerStatusEventData(entry, status);
            EventSystem.CallEvent(EventType.RENDERER_SERVER_STATUS, eventData);
        }

        if (this.status == ConnectionStatus.ERROR) return;
        this.status = status;
    }
}

File: Editor/src/main/kotlin/turboedit/editor/renderer/RenderServerConnection.kt
package turboedit.editor.renderer

import turboedit.editor.events.EventSystem.callEvent
import turboedit.editor.events.EventType
import turboedit.editor.events.RendererServerStatusEventData

class RenderServerConnection(private val entry: RenderServerEntry, private var client: RendererClient?) {
    private var status: ConnectionStatus? = ConnectionStatus.CONNECTING

    fun entry(): RenderServerEntry {
        return entry
    }

    fun client(): RendererClient? {
        return client
    }

    fun client(client: RendererClient?) {
        this.client = client
    }

    fun status(): ConnectionStatus? {
        return status
    }

    @JvmOverloads
    fun status(status: ConnectionStatus, callEvent: Boolean = true) {
        if (callEvent) {
            val eventData = RendererServerStatusEventData(entry, status)
            callEvent(EventType.RENDERER_SERVER_STATUS, eventData)
        }

        if (this.status == ConnectionStatus.ERROR) return

        this.status = status
    }
}
